### PR TITLE
Add Sql Server OPENJSON function

### PR DESCRIPTION
**Pull request by viceroypenguin**

### CODE REVIEW

File: Source/LinqToDB/DataProvider/SqlServer/SqlFn.cs [hunk 1]
```diff
@@ -1684,6 +1683,42 @@ public static int IsDate(string expression)
...
+		/// <summary>
+		/// <para><b><see href="https://docs.microsoft.com/en-us/sql/t-sql/functions/OPENJSON-transact-sql">OPENJSON (Transact-SQL)</see></b></para>
+		/// <para>A table-valued function that parses JSON text and returns objects and properties from the JSON input as rows and columns.</para>
+		/// </summary>
+		/// <param name="json">An expression. Typically the name of a variable or a column that contains JSON text.</param>
+		/// <returns>Returns a rowset view over the elements of a JSON object or array.</returns>
+		/// <exception cref="InvalidOperationException" />
+		/// <remarks>Only available on SQL Server 2016 or later, and compatibility mode for the database must be set to 130 or higher</remarks>
+		[Sql.TableExpression("OPENJSON({2}) {1}")]
+		public static ITable<JsonData> OpenJson(string? json)
```
Comment (sdanyliv) [inline Source/LinqToDB/DataProvider/SqlServer/SqlFn.cs:1702]: I think it is better to use the following technique: https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/DataProvider/PostgreSQL/PostgreSQLExtensions.cs#L360

File: Source/LinqToDB/DataProvider/SqlServer/SqlFn.cs [hunk 2]
```diff
@@ -1684,6 +1686,85 @@ public static int IsDate(string expression)
...
```
Comment (sdanyliv) [inline Source/LinqToDB/DataProvider/SqlServer/SqlFn.cs:1705]:
```suggestion
		public static IQueryable<JsonData> OpenJson(string? json)
```

File: Source/LinqToDB/DataProvider/SqlServer/SqlFn.cs [hunk 3]
```diff
@@ -1684,6 +1686,85 @@ public static int IsDate(string expression)
...
+		/// <para><b><see href="https://docs.microsoft.com/en-us/sql/t-sql/functions/OPENJSON-transact-sql">OPENJSON (Transact-SQL)</see></b></para>
+		/// <para>A table-valued function that parses JSON text and returns objects and properties from the JSON input as rows and columns.</para>
+		/// </summary>
+		/// <param name="json">An expression. Typically the name of a variable or a column that contains JSON text.</param>
+		/// <param name="path">A JSON path expression that specifies the property to query.</param>
+		/// <returns>Returns a rowset view over the elements of a JSON object or array.</returns>
+		/// <exception cref="InvalidOperationException" />
+		/// <remarks>Only available on SQL Server 2016 or later, and compatibility mode for the database must be set to 130 or higher</remarks>
+		[Sql.TableExpression("OPENJSON({2}, {3}) {1}")]
+		public static ITable<JsonData> OpenJson(string? json, string path)
```
Comment (sdanyliv) [inline Source/LinqToDB/DataProvider/SqlServer/SqlFn.cs:1720]:
```suggestion
		public static IQueryable<JsonData> OpenJson(string? json, string path)
```

**Comment by sdanyliv** [comment]
I cannot test on my environment. But if would like to see tests which uses `OpenJson` and uses returned `IQueryable<JsonData>` with `Where` and `Select`

**Comment by viceroypenguin** [comment]
> I cannot test on my environment. But if would like to see tests which uses `OpenJson` and uses returned `IQueryable<JsonData>` with `Where` and `Select`

Sure - just because I've been using it for several years doesn't mean it's going to work when we make more changes tomorrow. Thanks for the heads up.

**Review by sdanyliv — APPROVED** [review]

**Review by Shane32 — APPROVED** [review]

**Pull request merged**